homeassistant/components/matter/event.py

```diff
@@ -116,18 +118,297 @@ def _on_matter_node_event(
             presses = (data.data or {}).get("totalNumberOfPressesCounted", 1)
             event_type = f"multi_press_{presses}"
         else:
-            event_type = EVENT_TYPES_MAP[data.event_id]
+            event_type = GENERIC_SWITCH_EVENT_TYPES_MAP[data.event_id]
 
         if event_type not in self.event_types:
             # this should not happen, but guard for bad things
             # some remotes send events that they do not report as supported (sigh...)
             return
 
+        # For each button interaction, generate two event. The first is a high-level event that enables triggering on any type of button interaction.
+        # This first event also fixes an automation issue in which the state change trigger only picks up the event if the prior event
+        # was different. By inserting the any_button_interaction, it ensures that that code never generates two of the same events in a row
+        # the second event is more specific and conveys the specific interaction event.
+        self._trigger_event("any_button_interaction", data.data)
+        self.async_write_ha_state()
         # pass the rest of the data as-is (such as the advanced Position data)
         self._trigger_event(event_type, data.data)
         self.async_write_ha_state()
 
 
+DoorLockFeature = clusters.DoorLock.Bitmaps.Feature
+DOOR_LOCK_EVENT_TYPES = {
+    # mapping from raw event id's to translation keys
```

Comment:
translation keys are id's so lowercase, snake_case

Reply:
I'll go back and fix all of this. It will take me a week (I'm currently traveling for work), but will let you know when I'm done.